presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -351,39 +402,63 @@ private QueryResult executeQueries(
             if (queryResult.getState() == State.SUCCESS) {
                 queryResult = executeQuery(gateway, username, password, query, query.getQuery(), timeout, sessionProperties);
             }
+
+            return queryResult;
         }
-        finally {
-            int retry = 0;
-            QueryResult tearDownResult;
-            do {
-                tearDownResult = tearDown(query, postQueryResults, postQuery -> executeQuery(gateway, username, password, query, postQuery, timeout, sessionProperties));
-                if (tearDownResult.getState() == State.SUCCESS) {
-                    break;
-                }
-                try {
-                    TimeUnit.MINUTES.sleep(1);
-                }
-                catch (InterruptedException e) {
-                    Thread.currentThread().interrupt();
-                    break;
-                }
-                retry++;
-            }
-            while (retry < teardownRetries);
-            // if teardown is not successful the query fails
-            if (tearDownResult.getState() != State.SUCCESS) {
-                if (tearDownResult.getException() != null) {
-                    if (setupResult.getException() != null) {
-                        tearDownResult.addSuppressed(setupResult.getException());
-                    }
-                    if (queryResult.getException() != null) {
-                        tearDownResult.addSuppressed(queryResult.getException());
-                    }
-                }
-                queryResult = tearDownResult;
-            }
+        catch (Exception e) {
+            executeTearDown(query, gateway, username, password, timeout, postQueryResults, teardownRetries);
+            throw e;
         }
-        return queryResult;
+    }
+
+    private QueryResult executeTearDown(
+            Query query,
+            String gateway,
+            String username,
+            String password,
+            Duration timeout,
+            List<QueryResult> postQueryResults,
+            int teardownRetries)
+    {
+        int attempt = 0;
+        QueryResult tearDownResult;
+        do {
+            tearDownResult = tearDown(
+                    query,
+                    postQueryResults,
+                    postQuery -> executeQuery(
+                            gateway,
+                            username,
+                            password,
+                            query,
+                            postQuery,
+                            timeout,
+                            sessionProperties));
+            if (tearDownResult.getState() == State.SUCCESS) {
+                break;
+            }
+            try {
+                TimeUnit.MINUTES.sleep(1);
+                log.info(format("Query teardown failed on attempt #%d, will sleep and retry", attempt));
```

Comment:
Always use `%s` in preference to a plain `%d` as it is safer

Reply:
`format()` is not needed for the logger as the logger supports formatting natively